### PR TITLE
Read team.leave

### PR DESCRIPTION
**Pull request by mlsteele**
Read `team.leave` chain links. No tests for now but I bet it's probably right.

### CODE REVIEW

**Comment by mlsteele** [comment]
This PR is fixed now

File: go/teams/chain.go [hunk 1]
```diff
@@ -665,7 +662,65 @@ func (t *TeamSigChainPlayer) addInnerLink(prevState *TeamSigChainState, link SCC
...
+		}
+		if team.Members != nil {
+			return res, errors.New("unexpected members")
+		}
+		if team.Parent != nil {
+			return res, errors.New("unexpected parent")
+		}
+		if team.Subteam != nil {
+			return res, errors.New("unexpected subteam")
+		}
```
Comment (oconnor663) [inline go/teams/chain.go]: I wonder if we could factor some of this field checking into a single function, so the different types don't need to repeat it? Not for this PR though.

Comment (mlsteele) [inline go/teams/chain.go]: Yeah :D I have a PR on top of this one for that already. https://github.com/keybase/client/pull/7160

File: go/teams/chain.go [hunk 2]
```diff
@@ -665,7 +662,65 @@ func (t *TeamSigChainPlayer) addInnerLink(prevState *TeamSigChainState, link SCC
...
+			// ok
+		default:
+			return res, fmt.Errorf("link signer does not have permission to leave: %v is a %v", oRes.signingUser, signerRole)
+		}
+
+		// The last owner of a team should not leave.
+		// But that's really up to them and the server. We're just reading what has happened.
+
+		res.newState = prevState.DeepCopy()
+		res.newState.UserLog.inform(oRes.signingUser, keybase1.TeamRole_NONE, oRes.outerLink.Seqno)
```
Comment (oconnor663) [inline go/teams/chain.go]: Will there be any long-lived references to `prevState`, after the new state gets returned?

Comment (mlsteele) [inline go/teams/chain.go]: No. But just below there is a call to `prevState.GetLatestPerTeamKey()`. I didn't want to alias `prevState` and `newState`. If you're on the hunt for an extra DeepCopy I guess https://github.com/keybase/client/blob/master/go/teams/chain.go#L310 is one

**Pull request merged**